### PR TITLE
Fixed a deadlock in the PoolBasedSequentialScheduledExecutorService

### PR DESCRIPTION
**Pull request by joerg1985**
I did some testing with the new `PoolBasedSequentialScheduledExecutorService` and ended in a deadlock.
The `ScheduledThreadPoolExecutor` used in the background does not spawn new threads as expected.

This PR will call `setCorePoolSize` to spawn new threads or to stop idle threads. 
Additionally this PR does allow a task to be directly executed in case the current threads is from the pool. 

Signed-off-by: Jörg Sautter <joerg.sautter@gmx.net>

### CODE REVIEW

**Comment by joerg1985** [comment]
@kaikreuzer it would be great if this would be included in 4.2.M4

File: bundles/org.openhab.core/src/main/java/org/openhab/core/common/PoolBasedSequentialScheduledExecutorService.java [hunk 1]
```diff
@@ -100,22 +120,24 @@ public PoolBasedSequentialScheduledExecutorService(ScheduledThreadPoolExecutor p
                     tail = empty;
                 }
             }
-        }, 2, 4, TimeUnit.SECONDS);
+        },
+                // avoid cleaners of promptly created instances to run at the same time
```
Comment (kaikreuzer) [inline bundles/org.openhab.core/src/main/java/org/openhab/core/common/PoolBasedSequentialScheduledExecutorService.java:124]: the indentation does not look right here.

**Review by kaikreuzer — COMMENTED** [review]
The logic is getting pretty complex, but I hope it resolves the deadlock as expected.
Just a small formatting issue to be fixed before merging.

File: bundles/org.openhab.core/src/main/java/org/openhab/core/common/PoolBasedSequentialScheduledExecutorService.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/common/PoolBasedSequentialScheduledExecutorService.java:124]: Weird - I guess this is some bug in the spotless formatter. Well, let's leave it as it is then!

**Pull request merged**